integration/benchnet2/automate/bootstrap_automation.go

```
@@ -0,0 +1,196 @@
+	NetworkPubKey string `json:"NetworkPubKey"`
+	StakingPubKey string `json:"StakingPubKey"`
+}
+
+// ReplacementData struct which contains all data replacements for templates
+type ReplacementData struct {
+	NodeID   string
+	ImageTag string
+}
+
+var DEFAULT_NODE_INFO_PATH = "../bootstrap/public-root-information/node-infos.pub.json"
+
+var ACCESS_TEMPLATE string = "access_template.yml"
+var COLLECTION_TEMPLATE string = "collection_template.yml"
+var CONSENSUS_TEMPLATE string = "consensus_template.yml"
+var EXECUTION_TEMPLATE string = "execution_template.yml"
+var VERIFICATION_TEMPLATE string = "verification_template.yml"
+var RESOURCES_TEMPLATE string = "resources_template.yml"
+var TEMPLATE_PATH string = "templates/"
+
+var DEFAULT_ACCESS_IMAGE string = "gcr.io/flow-container-registry/access:v0.27.6"
+var DEFAULT_COLLECTION_IMAGE string = "gcr.io/flow-container-registry/collection:v0.27.6"
+var DEFAULT_CONSENSUS_IMAGE string = "gcr.io/flow-container-registry/consensus:v0.27.6"
+var DEFAULT_EXECUTION_IMAGE string = "gcr.io/flow-container-registry/execution:v0.27.6"
+var DEFAULT_VERIFICATION_IMAGE string = "gcr.io/flow-container-registry/verification:v0.27.6"
+
+func GenerateValuesYaml(inputJsonFilePath string, templatePath string, outputYamlFilePath string) {
+	templateFolder := TEMPLATE_PATH
+	if templatePath != "" {
+		templateFolder = templatePath
+	}
+	nodeInfoJson := DEFAULT_NODE_INFO_PATH
+	if inputJsonFilePath != "" {
+		nodeInfoJson = inputJsonFilePath
+	}
+	nodesData, nodeConfig := loadNodeJsonData(nodeInfoJson)
+
+	values := buildValuesStruct(templateFolder, nodesData, nodeConfig)
+
+	marshalToYaml(values, outputYamlFilePath)
+}
+
+// Loads node info json file and returns unmarshaled struct and node counts
+func loadNodeJsonData(nodeInfoPath string) (map[string]Node, map[string]int) {
+	jsonFileBytes := textReader(nodeInfoPath)
+
+	var nodes []Node
+	err := json.Unmarshal(jsonFileBytes, &nodes)
+	if err != nil {
+		log.Fatal(err)
```

We need to return this error and handle it.